force-app/main/default/classes/ExchangeRatesController.cls

```
@@ -0,0 +1,335 @@
+            
+            // Call the exchange rate provider
+            ExchangeRateProvider.ExchangeRateResult result = ExchangeRateProvider.fetchAndProcessExchangeRates();
+            
+            if (result.success) {
+                this.statusMessage = String.format('Successfully updated {0} exchange rates from CNB',
+                new List<Object>{result.processedCount});
+                setStatusMessage(STATUS_SUCCESS);
+                
+                // Reload the rates
+                loadExchangeRates();
+            } else {
+                this.statusMessage = 'Failed to update exchange rates: ' + result.message;
+                setStatusMessage(STATUS_ERROR);
+                
+                // Add detailed error information
+                if (!result.errors.isEmpty()) {
+                    this.statusMessage += '\nErrors: ' + String.join(result.errors, '; ');
+                }
+            }
+            
+        } catch (Exception e) {
+            this.statusMessage = 'Error updating exchange rates: ' + e.getMessage();
+            setStatusMessage(STATUS_ERROR);
+            System.debug(LoggingLevel.ERROR, 'Error in updateFromCNB: ' + e.getMessage());
+            System.debug(LoggingLevel.ERROR, 'Stack trace: ' + e.getStackTraceString());
+        } finally {
+            this.isLoading = false;
+        }
+    }
+    
+    /**
+     * @description Resets filters to default values
+     */
+    public void resetFilters() {
+        this.selectedDate = Date.today();
+        this.currencyFilter = '';
+        loadExchangeRates();
+    }
+    
+    /**
+     * @description Loads scheduler information
+     */
+    public void loadSchedulerInfo() {
+        try {
+            this.scheduledJobs = ExchangeRateScheduler.getScheduledJobInfo();
+        } catch (Exception e) {
+            System.debug(LoggingLevel.ERROR, 'Error loading scheduler info: ' + e.getMessage());
```

🛑 **[Robustness & UX] Don't Hide Backend Errors from the User**

This silent `catch` block in `loadSchedulerInfo` is going to create a confusing experience for users. If the backend call fails (for example, due to permissions), the exception is swallowed and the page will simply display "No scheduled jobs found," which isn't true. The user will have no idea that an error actually occurred.

Instead of hiding the failure, the `catch` block should update the `statusMessage` property with a helpful error. This ensures the user is always informed about what's happening. This change works together with the recommendation for the backend to throw exceptions instead of handling them silently.